src/tests/ftest/util/ServerUtils.py

```diff
@@ -33,18 +33,126 @@
+    """
+    if 'bdev_class' in default_yaml_value['servers'][0]:
+        if default_yaml_value['servers'][0]['bdev_class'] == bdev and not enabled:
+            del default_yaml_value['servers'][0]['bdev_class']
+    if enabled:
+        default_yaml_value['servers'][0]['bdev_class'] = bdev
+
+def remove_mux_from_yaml(avocado_yaml_file, avocado_yaml_file_tmp):
+    """
+    Function to remove "!mux" from yaml file and create new tmp file
+    Args:
+        avocado_yaml_file: Avocado test yaml file
+        avocado_yaml_file_tmp: Avocado temporary test yaml file
+    """
+    with open(avocado_yaml_file, 'r') as rfile:
+        filedata = rfile.read()
+    filedata = filedata.replace('!mux', '')
+    with open(avocado_yaml_file_tmp, 'w') as wfile:
+        wfile.write(filedata)
+
+def update_server_options(avocado_yaml_value, default_yaml_value):
+    """
+    Update the 'server_options' from avocado_yaml_value dictionary to default_yaml_value dictionary.
+    """
+    default_yaml_env = default_yaml_value["servers"][0]['env_vars']
+
+    if 'server_options' in avocado_yaml_value:
+        avocado_yaml_options = avocado_yaml_value['server_options']
+        #Iterate through the server_options value from test file
+        for key, val in avocado_yaml_options.iteritems():
+            if 'env_vars' in key:
+                for no_of_env in enumerate(val):
+                    for env_k, env_v in no_of_env[1].iteritems():
+                        env_index = [default_yaml_env.index(i)
+                                     for i in  default_yaml_env if env_k in i]
+                        #Update the Existing environment variable
+                        if env_index:
+                            default_yaml_env[env_index[0]] = "{}={}".format(env_k, env_v)
+                        #Add new environment variable
+                        else:
+                            default_yaml_env.append("{}={}".format(env_k, env_v))
+            #Update/Add other non env_vars variable
+            else:
+                default_yaml_value["servers"][0].update({key:val})
+
+def create_server_yaml(basepath):
+    """
+    This function is to create DAOS server configuration YAML file based on Avocado test Yaml file.
+    Args:
+        - basepath = DAOS install basepath
+    """
+    #Read the default utils/config/examples/daos_server_sockets.yml
+    try:
+        with open('{}/{}'.format(basepath, DEFAULT_YAML_FILE), 'r') as read_file:
+            default_yaml_value = yaml.safe_load(read_file)
+    except Exception as excpn:
+        print("<SERVER> Exception occurred: {0}".format(str(excpn)))
+        traceback.print_exception(excpn.__class__, excpn, sys.exc_info()[2])
+        raise ServerFailed("Failed to Read {}/{}".format(basepath, DEFAULT_YAML_FILE))
+
+    #Read the values from avocado_testcase.yaml file if test ran with Avocado.
+    avocado_yaml_value = ""
+    if "AVOCADO_TEST_DATADIR" in os.environ:
```

Comment:
Some other options would be to pass the yaml file name as a parameter or to read the test yaml in the test and load the config file relavent values as a dictionary into this function.

Reply:
This has been updated. If user run this test without Avocado it will use the default yaml file and start the server.